providers/implementations/ciphers/ciphercommon_ccm.c

```diff
@@ -74,6 +74,19 @@ int ossl_ccm_set_ctx_params(void *vctx, const OSSL_PARAM params[])
     if (params == NULL)
         return 1;
 
+    p = OSSL_PARAM_locate_const(params, OSSL_CIPHER_PARAM_AEAD_TAGLEN);
```

Comment:
Hmm I deliberately made this only a getter when I added it..
(Since the ctrls had to work this way - i.e ctrl can set the tag to NULL but still specify a length..)

If we want to handle this this way then quite a few more things need to change (including the provider-cipher.doc) and ossl_cipher_aead_settable_ctx_params() which is also shared with aes_gcm + aria_ccm and aria_gcm.

Reply:
> Hmm I deliberately made this only a getter when I added it..
> (Since the ctrls had to work this way - i.e ctrl can set the tag to NULL but still specify a length..)
> 

That would be fine, but then a developer comes along and assumes they can set the tag length in a parameter as there's nothing to state otherwise, and no error returned! ;)

Reply:
It is documented

=item "taglen" (B<OSSL_CIPHER_PARAM_AEAD_TAGLEN>) <unsigned integer>

Gets the tag length to be used for an AEAD cipher for the associated cipher ctx.
It gets a default value if it has not been set.
The length of the "taglen" parameter should not exceed that of a B<size_t>.

And it is also just a gettable param..
Potentially it could return 0 if you try to set it here.

Reply:
This change has been reverted